src/sentry/snuba/metrics.py

```diff
@@ -1112,16 +1077,167 @@ def get_series(self, projects: Sequence[Project], query: QueryDefinition) -> dic
+
+            snuba_queries = SnubaQueryBuilder(projects, query).get_snuba_queries()
+            if len(snuba_queries) > 1:
+                # Currently accepting an order by field that spans multiple entities is not
+                # supported, but it might change in the future. Even then, it might be better
+                # handled on the snuba side of things
+                raise InvalidParams(
+                    "Order by queries over multiple entities are not supported in "
+                    "multi-field select with order by clause queries"
+                )
+
+            # This query contains an order by clause, and so we are only interested in the
+            # "totals" query
+            initial_snuba_query = next(iter(snuba_queries.values()))["totals"]
+            # ToDo(ahmed): Change this to accept the limit from the api request with a fallback to
+            #  50 elements if more than 50 elements are to be requested
+            initial_snuba_query = initial_snuba_query.set_limit(50)
+
+            initial_query_results = raw_snql_query(
+                initial_snuba_query, use_cache=False, referrer="api.metrics.totals"
+            )
+
+            # We no longer want the order by in the 2nd query because we already have the order of
+            # the group by tags from the first query so we basically remove the order by columns,
+            # and reset the query fields to the original fields because in the second query,
+            # we want to query for all the metrics in the request api call
+            query.orderby = None
+            query.fields = original_query_fields
+
+            snuba_queries = SnubaQueryBuilder(projects, query).get_snuba_queries()
+
+            results = {entity: {"totals": {"data": []}} for entity in snuba_queries.keys()}
+
+            # If we do not get any results from the first query, then there is no point in making
+            # the second query
+            if len(initial_query_results["data"]) > 0:
+                # Translate the groupby fields of the query into their tag keys because these fields
+                # will be used to filter down and order the results of the 2nd query.
+                # For example, (project_id, transaction) is translated to (project_id, tags[3])
+                groupby_tags = [
+                    resolve_tag_key(field) if field not in _ALLOWED_GROUPBY_COLUMNS else field
+                    for field in query.groupby
+                ]
+
+                # Dictionary that contains the conditions that are required to be added to the where
+                # clause of the second query. In addition to filtering down on the tuple combination
+                # of the fields in the group by columns, we need a separate condition for each of
+                # the columns in the group by with their respective values so Clickhouse can
+                # filter the results down before checking for the group by column combinations.
```

Comment:
Out of curiosity, how did you find out that you also need the single columns as conditions?

Reply:
I talked with Evan about this, shared with him the query and he made that suggestion